### PR TITLE
homework number ten

### PR DESCRIPTION
**Pull request by kihors**

### CODE REVIEW

File: src/homeworkTenth/MyHashMap.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        this(INITIAL_CAPACITY);
+    }
+
+    public MyHashMap(int capacity) {
+        this.buckets = new Entry[capacity];
+    }
+
+    public void put(K key, V value) {
+        Entry<K, V> entry = new Entry<>(key, value, null);
+        int bucket = getHash(key) % getBucketSize();
```
Comment (SevostyanovGerman) [inline src/homeworkTenth/MyHashMap.java]: cs

File: src/homeworkTenth/MyHashMap.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        Entry<K, V> exiting = buckets[bucket];
+        if (exiting == null) {
+            buckets[bucket] = entry;
+            size++;
+        } else {
+            while (exiting.next != null) {
```
Comment (SevostyanovGerman) [inline src/homeworkTenth/MyHashMap.java]: may be change with second condition?

File: src/homeworkTenth/MyHashMap.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    public V get(K key) {
+        Entry<K, V> bucket = buckets[getHash(key) % getBucketSize()];
+
+        while (bucket != null) {
+            if (key == bucket.key) {
+                return bucket.value;
+            }
+            bucket = bucket.next;
+        }
+        return null;
```
Comment (SevostyanovGerman) [inline src/homeworkTenth/MyHashMap.java]: ??

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/homeworkTenth/MyHashMap.java (hunk 3 above)
Comment (kihors) [reply src/homeworkTenth/MyHashMap.java]: we are computing the hash code from the key, and then compute the index from the hash code with module operation. 
Then, get the linked list at index computed above and search through the linked list for the value with this value.

File: src/homeworkTenth/MyHashMap.java (hunk 2 above)
Comment (kihors) [reply src/homeworkTenth/MyHashMap.java]: no. 
here we compare the keys if the key already exists

**Pull request merged**